### PR TITLE
Yolov3 loss: add mixup score and label smooth

### PR DESCRIPTION
**Pull request by heavengate**
红框中为新增文档

3ddb4deb45bbbe56793b1c4cb

### CODE REVIEW

**Comment by heavengate** [comment]
@haowang101779990  please check document

**Comment by heavengate** [comment]
@haowang101779990 please check doc

**Review by jerrywgz — DISMISSED** [review]
LGTM

File: paddle/fluid/operators/detection/yolov3_loss_op.cc [hunk 1]
```diff
@@ -204,6 +225,15 @@ class Yolov3LossOpMaker : public framework::OpProtoAndCheckerMaker {
...
               + loss_{conf} + loss_{class}
          $$
+
+         While :attr:`use_label_smooth` is set to be :attr:`True`, the classification
+         target will be smoothed when calculating classification loss, target of 
+         positive samples will be smoothed to :math:`1.0 - 1.0 / class\_num` and target of
+         negetive samples will be smoothed to :math:`1.0 / class\_num`.
+
+         While :attr:`GTScore` is given, which means the mixup score of ground truth 
+         boxes, all looses incured by a ground truth box will be multiplied by its
```
Comment (haowang101779990) [inline paddle/fluid/operators/detection/yolov3_loss_op.cc]:
```suggestion
         boxes, all losses incured by a ground truth box will be multiplied by its 
```

Comment (heavengate) [reply paddle/fluid/operators/detection/yolov3_loss_op.cc]: Done

**Review by shanyi15 — DISMISSED** [review]
LG comment

**Comment by shanyi15** [comment]
建议参数说明name那儿再说一次默认值

**Review by shanyi15 — DISMISSED** [review]
LG comments

File: paddle/fluid/operators/detection/yolov3_loss_op.h [hunk 2]
```diff
@@ -283,6 +297,19 @@ class Yolov3LossKernel : public framework::OpKernel<T> {
     int* gt_match_mask_data =
         gt_match_mask->mutable_data<int>({n, b}, ctx.GetPlace());
 
+    const T* gt_score_data;
+    if (!gt_score) {
+      Tensor _gt_score;
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/yolov3_loss_op.h]: _gt_score  -> gt_score , do not add _ prefix

File: python/paddle/fluid/layers/detection.py [hunk 3]
```diff
@@ -515,7 +515,9 @@ def yolov3_loss(x,
                 class_num,
                 ignore_thresh,
                 downsample_ratio,
-                name=None):
+                name=None,
+                gtscore=None,
+                use_label_smooth=True):
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]:
```
gtscore=None,
use_label_smooth=True,
name=None):
```

File: python/paddle/fluid/layers/detection.py [hunk 4]
```diff
@@ -533,28 +535,35 @@ def yolov3_loss(x,
...
 
     Examples:
       .. code-block:: python
 
           x = fluid.layers.data(name='x', shape=[255, 13, 13], dtype='float32')
-          gtbox = fluid.layers.data(name='gtbox', shape=[6, 5], dtype='float32')
-          gtlabel = fluid.layers.data(name='gtlabel', shape=[6, 1], dtype='int32')
+          gtbox = fluid.layers.data(name='gtbox', shape=[6, 4], dtype='float32')
+          gtlabel = fluid.layers.data(name='gtlabel', shape=[6], dtype='int32')
+          gtscore = fluid.layers.data(name='gtscore', shape=[6], dtype='int32')
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: The type soc gtscore is int3 ?

Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: The type soc gtscore is int32 ?

File: python/paddle/fluid/layers/detection.py [hunk 5]
```diff
@@ -566,6 +575,8 @@ def yolov3_loss(x,
         raise TypeError("Input gtbox of yolov3_loss must be Variable")
     if not isinstance(gtlabel, Variable):
         raise TypeError("Input gtlabel of yolov3_loss must be Variable")
+    if not isinstance(gtscore, Variable):
```
Comment (qingqing01) [inline python/paddle/fluid/layers/detection.py]: gtscore不可以None?

File: paddle/fluid/operators/detection/yolov3_loss_op.h [hunk 6]
```diff
@@ -414,25 +451,41 @@ class Yolov3LossGradKernel : public framework::OpKernel<T> {
         input_grad->mutable_data<T>({n, c, h, w}, ctx.GetPlace());
     memset(input_grad_data, 0, input_grad->numel() * sizeof(T));
 
+    const T* gt_score_data;
+    if (!gt_score) {
+      Tensor _gt_score;
+      _gt_score.mutable_data<T>({n, b}, ctx.GetPlace());
```
Comment (qingqing01) [inline paddle/fluid/operators/detection/yolov3_loss_op.h]: save as above

File: paddle/fluid/operators/detection/yolov3_loss_op.h (hunk 2 above)
Comment (heavengate) [reply paddle/fluid/operators/detection/yolov3_loss_op.h]: Done

File: paddle/fluid/operators/detection/yolov3_loss_op.h (hunk 6 above)
Comment (heavengate) [reply paddle/fluid/operators/detection/yolov3_loss_op.h]: Done

File: python/paddle/fluid/layers/detection.py (hunk 3 above)
Comment (heavengate) [reply python/paddle/fluid/layers/detection.py]: Done

File: python/paddle/fluid/layers/detection.py (hunk 4 above)
Comment (heavengate) [reply python/paddle/fluid/layers/detection.py]: Done

Comment (heavengate) [reply python/paddle/fluid/layers/detection.py]: Done

File: python/paddle/fluid/layers/detection.py (hunk 5 above)
Comment (heavengate) [reply python/paddle/fluid/layers/detection.py]: Done

**Review by qingqing01 — DISMISSED** [review]

**Review by qingqing01 — APPROVED** [review]

**Review by qingqing01 — APPROVED** [review]

**Review by shanyi15 — APPROVED** [review]
LG comments

**Review by XiaoguangHu01 — APPROVED** [review]

**Pull request merged**